core/src/main/scala/dagr/core/execsystem/TaskExecutionInfo.scala

```diff
@@ -42,7 +42,6 @@ import dagr.core.tasksystem.Task
  * @param attemptIndex the one-based count of attempts to run this task.
  */
 class TaskExecutionInfo(var task: Task,
-                        val id: BigInt,
                         var status: TaskStatus.Value,
```

Comment:
I see.  I think putting a reference to `TaskExecutionInfo` into `Task` makes sense since it would simplify more of the tracking/book-keeping.  The other option would be to use `TaskExecutionInfo` _everywhere_ in TaskManager and friends, and have it be the _only_ way to get at a _Task_, so that it's basically a super-wrapper around the task inside the execution/management world.

I find what I really want is post-creation ad-hoc mixins:
`val superTask = myTask with TaskExecutionInfo`
so that you could augment an _instance_ you already have with another trait.